tests/pipeline/test_dividends.py

```diff
@@ -0,0 +1,496 @@
+                          'NaT'],
+                         ['NaT', '2014-01-10', 'NaT', '2014-01-20', 'NaT'],
+                         ['NaT', '2014-01-10', '2014-01-15', 'NaT']]
+
+prev_ex_and_pay_dates = [['NaT', '2014-01-15', '2014-01-20'],
+                         ['NaT', '2014-01-15', '2014-01-20'],
+                         ['NaT', '2014-01-10', '2014-01-20'],
+                         ['NaT', '2014-01-10', '2014-01-15']]
+
+prev_amounts = [['NaN', 1, 15],
+                ['NaN', 13, 7],
+                ['NaN', 3, 1],
+                ['NaN', 6, 23]]
+
+next_amounts = [['NaN', 1, 15, 'NaN'],
+                ['NaN', 7, 13, 7, 'NaN'],
+                ['NaN', 3, 'NaN', 1, 'NaN'],
+                ['NaN', 6, 23, 'NaN']]
+
+
+def get_vals_for_dates(zip_date_index_with_vals,
+                       vals,
+                       date_invervals,
+                       dates):
+    return pd.DataFrame({
+        0: get_values_for_date_ranges(zip_date_index_with_vals,
+                                      vals[0],
+                                      date_invervals[0],
+                                      dates),
+        1: get_values_for_date_ranges(zip_date_index_with_vals,
+                                      vals[1],
+                                      date_invervals[1],
+                                      dates),
+        2: get_values_for_date_ranges(zip_date_index_with_vals,
+                                      vals[2],
+                                      date_invervals[2],
+                                      dates),
+        # Assume the latest of 2 cash values is used if we find out about 2
+        # announcements that happened on the same day for the same sid.
+        3: get_values_for_date_ranges(zip_date_index_with_vals,
+                                      vals[3],
+                                      date_invervals[3],
+                                      dates),
+        4: zip_date_index_with_vals(dates, ['NaN'] * len(dates)),
+    }, index=dates)
+
+
+class DividendsByAnnouncementDateTestCase(TestCase, EventLoaderCommonMixin):
+    """
+    Tests for loading the dividends by announcement date data.
+    """
+    pipeline_columns = {
+        PREVIOUS_ANNOUNCEMENT:
+            DividendsByAnnouncementDate.previous_announcement_date.latest,
+        PREVIOUS_AMOUNT: DividendsByAnnouncementDate.previous_amount.latest,
+        DAYS_SINCE_PREV_DIVIDEND_ANNOUNCEMENT:
+            BusinessDaysSinceDividendAnnouncement(),
+    }
+
+    @classmethod
+    def get_sids(cls):
+        return range(0, 5)
+
+    @classmethod
+    def setUpClass(cls):
+        cls._cleanup_stack = stack = ExitStack()
+        cls.cols = {}
+        cls.dataset = {sid:
+                       frame.drop([EX_DATE_FIELD_NAME,
+                                   PAY_DATE_FIELD_NAME], axis=1)
+                       for sid, frame
+                       in enumerate(dividends_cases)}
+        cls.finder = stack.enter_context(
```

Comment:
We also need to change `test_buyback_auth.py` and `test_earnings.py` to use that, right?